src/rgw/rgw_sal_motr.cc

```diff
@@ -137,6 +137,25 @@ static uint64_t rounddown(uint64_t x, uint64_t by)
   return x / by * by;
 }
 
+int parse_tags(const DoutPrefixProvider* dpp, bufferlist& tags_bl, struct req_state* s)
+{
+  std::unique_ptr<RGWObjTags> obj_tags;
+  if (s->info.env->exists("HTTP_X_AMZ_TAGGING")) {
+    auto tag_str = s->info.env->get("HTTP_X_AMZ_TAGGING");
+    obj_tags = std::make_unique<RGWObjTags>();
+    int ret = obj_tags->set_from_string(tag_str);
+    if (ret < 0) {
+      ldpp_dout(dpp, 0) <<__func__<< "setting obj tags failed with rc=" << ret << dendl;
+      if (ret == -ERR_INVALID_TAG) {
+        ret = -EINVAL; //s3 returns only -EINVAL for PUT requests
+      }
+      return ret;
+    }
+    obj_tags->encode(tags_bl);
+  }
```

Comment:
do we need to handle negative scenario here .
e.g. if this tag header (HTTP_X_AMZ_TAGGING) is missing then do we need to return error code or this parse_tag() method will still return success code ?

Reply:
if there are no tags in requests, then function should return 0, given there was no error

Reply:
Yes tagging is optional for copy-object